### PR TITLE
#87: [modal] implement modal manager (closes #87)

### LINKED ISSUE #87: [modal] implement modal manager
## requirements
simple stateless modal manager to switch between available modals and show only one at a time. it should provide a basic animation for entering/leaving modals

## specs
- `ModalManager` should be stateless: the parent should pass it the data it needs through `props`
- renders `null` and create + populate `div` outside main tree with current `Modal`(if any)
- accepts three props:
  - `activeModal: t.maybe(t.Str) // the uniqueID of the active modal`
  - `data: t.maybe(t.Obj) // the data passed to the modal as {...data}`
  - `modals: t.list( t.struct( { id: t.Str, modal: t.ReactElement } ) )`
- semantic free
  - desktop
  - mobile (copy from OXWAY?)
- basic enter/leave animations (custom animation support will be added in future)

### PR DESCRIPTION
**Pull request by FrancescoCioria**
Issue #87

### CODE REVIEW

File: examples/examples.js [hunk 1]
```diff
@@ -31,7 +33,7 @@ const template = (
 );
 
 document.getElementById('iso-container').innerHTML = React.renderToString(template);
-React.render(template, document.getElementById('iso-container'));
+// React.render(template, document.getElementById('iso-container'));
```
Comment (giogonzo) [inline examples/examples.js]: maybe open an issue to make this iso-friendly?

Comment (FrancescoCioria) [inline examples/examples.js]: it is already, only for dev experience I wanted only one modal open at a time. This wasn't supposed to be committed.. reverting

Comment (giogonzo) [inline examples/examples.js]: :+1:

On Thu, Nov 12, 2015 at 10:44 AM, Francesco <notifications@github.com>
wrote:

> In examples/examples.js
> <https://github.com/buildo/react-components/pull/88#discussion_r44637424>:
>
> > @@ -31,7 +33,7 @@ const template = (
> >  );
> >
> >  document.getElementById('iso-container').innerHTML = React.renderToString(template);
> > -React.render(template, document.getElementById('iso-container'));
> > +// React.render(template, document.getElementById('iso-container'));
>
> it is already, only for dev experience I wanted only one modal open at a
> time. This wasn't supposed to be committed.. reverting
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/buildo/react-components/pull/88/files#r44637424>.
>



-- 
Giovanni
~ buildo

**Comment by FrancescoCioria** [comment]
don't merge this yet as I discovered some issues about `onClickOutside`. I think it could be solved by using the new component `BackgroundDimmer`

**Comment by giogonzo** [comment]
Ok, seems way to complex to be reviewed normally (especially in the arbitrary setTimeout part :P).
I'd suggest we just follow the usual flow and merge when needed, test on projects, fix what's missing

**Comment by FrancescoCioria** [comment]
@giogonzo I fixed the issue. I'm merging this

**Comment by giogonzo** [comment]
:+1:

On Thu, Nov 12, 2015 at 11:51 AM, Francesco <notifications@github.com>
wrote:

> @giogonzo <https://github.com/giogonzo> I fixed the issue. I'm merging
> this
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/buildo/react-components/pull/88#issuecomment-156071168>
> .
>



-- 
Giovanni
~ buildo

**Pull request merged**